tests/test_interface.rb

```diff
@@ -528,43 +564,68 @@ def test_mtu_invalid_loopback
 
   def test_speed
     interface = Interface.new(interfaces[0])
-    if platform == :ios_xr
+    if node.product_id =~ /C3132/
+      skip('Speed is not supported on this platform')
+    elsif validate_property_excluded?('interface', 'speed')
```

Comment:
I mistakenly thought the issue with `nil` values being turned into `0` by `val.to_i` was an issue with C3132 platforms.  It's removed now.